lorien/UI/Dialogs/AddKeyDialog.gd

```diff
@@ -0,0 +1,77 @@
+extends WindowDialog
+
+# -------------------------------------------------------------------------------------------------
+const _MODIFIER_KEYS := [KEY_SUPER_L, KEY_SUPER_R, KEY_CONTROL, KEY_SHIFT, KEY_META, KEY_ALT]
+
+# -------------------------------------------------------------------------------------------------
+export var action_name := ""
+export var readable_action_name := "" setget _set_readable_action_name
+
+onready var _confirm_rebind_dialog := $ConfirmRebind
+
+var _pending_bind_event = null
+
+# -------------------------------------------------------------------------------------------------
+func _ready() -> void:
+	_update_event_text()
+	GlobalSignals.connect("language_changed", self, "_update_event_text")
+
+# -------------------------------------------------------------------------------------------------
+func _set_readable_action_name(s: String):
+	readable_action_name = s
+	_update_event_text()
+
+# -------------------------------------------------------------------------------------------------
+func _update_event_text() -> void:
+	$VBoxContainer/EventText.text = tr(
+		"KEYBINDING_DIALOG_BIND_ACTION"
+	).format({"action": readable_action_name})
+
+# -------------------------------------------------------------------------------------------------
+func _action_for_event(event: InputEvent):
+	for action in Utils.bindable_actions():
+		if InputMap.action_has_event(action, event):
+			return action
+	return null
+
+# -------------------------------------------------------------------------------------------------
+func _input(event: InputEvent) -> void:
+	if ! visible || _confirm_rebind_dialog.visible:
+		return
+	if event is InputEventKey && event.is_pressed():
+		get_tree().set_input_as_handled()
+		
+		if event.scancode in _MODIFIER_KEYS:
```

Comment:
Had to change this. The mask-based detection I used before was misguided - I interpreted the mask-value incorrectly. This is more specific and works correctly. In the previous version, you were not able to, for example, bin Escape... this fixes that problem...

Reply:
Alrighty 👍